src/Common/src/Interop/Unix/System.Native/Interop.UName.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Get the name of the current system. 
+        /// </summary>
+        /// <param name="machine">The OS architecture value.</param>
+        /// <returns>
+        /// Returns non-negative value on success, -1 on failure.
+        /// </returns>
+        [DllImport(Libraries.SystemNative, CharSet = CharSet.Ansi, SetLastError = true)]
+        internal static extern int UName([Out] StringBuilder machine, int capacity);
+
+        internal static int UName(out string machine)
+        {
+            int maxBufferLength = 1024;
+            StringBuilder buffer = new StringBuilder(maxBufferLength);
```

Comment:
There's no reason we need to be using StringBuilder here.  We can just allocate a char[] (or even stackalloc it) and then use Marshal.PtrToStringAnsi to get the result out.  That'll be much more efficient than using a StringBuilder, given how StringBuilder marshaling currently works.